.pre-commit-hooks.yaml

```diff
@@ -65,3 +65,27 @@
     entry: "\uFFFD"
     language: pygrep
     types: [text]
+-   id: ts-no-debugger
+    name: check for debbuger
+    description: 'A quick check for the `debugger` statement'
+    language: pygrep
+    entry: '\bdebugger\b'
+    types: [ts]
+-   id: ts-no-console
+    name: check console functions
+    description: 'A quick check for the use of `console` built-in functions'
+    language: pygrep
+    entry: '\bconsole\.'
+    types: [ts]
+-   id: ts-no-window
+    name: check window access
+    description: 'A quick check for the use of the `window` object'
+    language: pygrep
+    entry: '\bwindow\b'
+    types: [ts]
+-   id: ts-no-focus-ignore
+    name: check use of focus and ignore of tests
+    description: 'A quick check for the use of `fit()`, `xit()`, `fdescribe()` and `xdescribe()` test functions'
```

Comment:
not sure why you put "A quick check" on all of these

Reply:
also this one in particular seems too specific, it's also unclear _why_ someone would use this one?

Reply:
If testing things out locally I've often seen these changed, and then checked in.
As a consequence, later on the focus/ignore things need to be removed again. Ignore is more dangerous, as it only removes that one test case, which might not be immediately obvious from the CI results.

Reply:
Actually, just to be clear: you can get checks with eslint, too, by using the `no-restricted-globals` rule.

Reply:
I still don't know what `fit` / `xit` / etc. do, could you explain them?

Reply:
Sure. `describe` is used to gather `it` functions, which are test cases. `f` variants lead to the test runner only executing that one block/test case, while `x` variants lead to the block/test case being ignored.